File: app/src/main/java/ca/fuwafuwa/kaku/Search/SearchInfo.java
package ca.fuwafuwa.kaku.Search;

import ca.fuwafuwa.kaku.Windows.Views.KanjiCharacterView;

/**
 * Created by Xyresic on 12/16/2016.
 */

public class SearchInfo {

    private final String text;
    private final int textOffset;
    private final KanjiCharacterView kanjiCharacterView;

    public SearchInfo(String text, int textOffset, KanjiCharacterView kanjiCharacterView){
        this.text = text;
        this.textOffset = textOffset;
        this.kanjiCharacterView = kanjiCharacterView;
    }

    public String getText() {
        return text;
    }

    public int getTextOffset() {
        return textOffset;
    }

    public KanjiCharacterView getKanjiCharacterView() {
        return kanjiCharacterView;
    }
}

File: app/src/main/java/ca/fuwafuwa/kaku/Search/SearchInfo.kt
package ca.fuwafuwa.kaku.Search

import ca.fuwafuwa.kaku.Windows.Data.DisplayData
import ca.fuwafuwa.kaku.Windows.Data.ISquareChar
import ca.fuwafuwa.kaku.Windows.Views.KanjiCharacterView

/**
 * Created by 0xbad1d3a5 on 12/16/2016.
 */

class SearchInfo(private val displayData: DisplayData, private val squareChar: ISquareChar)
{
    val text: String get() = displayData.text

    val textOffset: Int get() = squareChar.index
}
